trlx/trainer/accelerate_base_trainer.py

```diff
@@ -308,6 +308,7 @@ def save_pretrained(self, directory: Optional[str] = None, **kwargs):
     def save(self, directory: Optional[str] = None, **kwargs):
         """Creates a checkpoint of the optimizer, scheduler and model"""
         self.accelerator.save_state(directory or self.config.train.checkpoint_dir, **kwargs)
+        self.accelerator.unwrap_model(self.model).config.save_pretrained(directory or self.config.train.checkpoint_dir)
```

Comment:
shouldn't we call self.save_pretrained here?

Reply:
This line is specifically to save the config file `self.accelerator.unwrap_model(self.model).`**config**`.save_pretrained`

I think `save_state` saves the model weights. 

`Save_pretrained` also saves the model weights, but I'm not sure how it would interact with the results from `save_state` - for example if it overwrites the whole dir that would be undesirable.

IME adding a config file to the save dir created by `save_state` is enough to load the model with `from_pretrained`. I'm assuming that adding the config file wouldn't interfere with `.load`